pandas/core/indexes/datetimelike.py

```diff
@@ -1137,3 +1133,29 @@ def _ensure_datetimelike_to_i8(other):
             # period array cannot be coerces to int
             other = Index(other).asi8
     return other
+
+
+def maybe_wrap_in_index(result, name=None, strict=True):
```

Comment:
`indexes.base` makes sense, sure.

The `strict` parameter is there because most of the use cases don't have the try/except for the index wrapping. `strict=False` is only passed for the comparison ops.  I'll see if I can come up with something prettier.  At least this has served its purpose of tagging the next few things to be moved.